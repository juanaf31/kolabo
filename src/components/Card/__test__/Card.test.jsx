import { cleanup, getByTestId, getByText, render } from '@testing-library/react'
import React from 'react'
import CardComponent from '../index'
import renderer from 'react-test-renderer'

afterEach(() => {
    cleanup()
})

describe('testing card component', () => {
    const val = 'anyValue'

    test('render components correctly', () => {
        const { getByTestId } = render(<CardComponent />)
        getByTestId('card_component')
    })

    test('render props correctly', () => {
        const { getByTestId } = render(<CardComponent title={val} className={val} />)

        getByTestId('card_component').hasAttribute('title')
        getByTestId('card_component').hasAttribute('className')
    })

    test('card matches snapshot', () => {
        const tree = renderer.create(<CardComponent />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})