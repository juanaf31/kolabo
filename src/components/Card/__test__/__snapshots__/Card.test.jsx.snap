// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing card component card matches snapshot 1`] = `
<div
  className="ant-card ant-card-bordered"
  data-testid="card_component"
  style={
    Object {
      "borderRadius": "10px",
      "width": "100%",
    }
  }
>
  <div
    className="ant-card-body"
    style={Object {}}
  />
</div>
`;
